readme.md

```diff
@@ -176,6 +170,37 @@ class Owner {
 getRootInjector().register(Owner, { metadata: Pet }).register(Pet)
 ```
 
+## Decorators & Metadata
+
+This library performs runtime introspection in order to determine what types it should construct.  To do this the library uses metadata and generally this metadata will be implicitly captured for you if you have enabled TypeScripts `emitDecoratorMetadata` and a class is decorated with any decorator. However, you do not need to use decorators or `emitDecoratorMetadata` if you do not wish to.  In that case you will need to manually provide the metadata using a decorator or registration API.  **Note** Managing the metadata explicitly can be time consuming so we **recommend using the auto generated metadata approach by default**.  
+
+```typescript
+import { Injectable } from '@morgan-stanley/needle';
+
+@Injectable()
+class Pet {}
+
+//Explicit metadata via the metadata property on the decorator. 
+@Injectable({ metadata: [Pet]})
+class Owner {
+    constructor(pet: Pet) {}
+}
+
+//Explicit metadata using the registration API 
+getRootInjector().register(Owner, { metadata: Pet }).register(Pet)
```

Comment:
metadata is inconsistent. Here it's a single value. In the other example on line184 it's a single item array. Is this intentional?

Reply:
no docs issue. Will fix